pkgs/by-name/di/dita-ot/package.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv
+, fetchzip
+, bash
+, jdk17
+, nix
+, coreutils
+, lib
+}:
+
+stdenv.mkDerivation rec {
+  name = "dita-ot";
+  version = "4.2.1";
+
+  buildInputs = [
+    bash
+    jdk17
+  ];
+
+  src = fetchzip {
+    url = "https://github.com/dita-ot/dita-ot/releases/download/${version}/dita-ot-${version}.zip";
+    hash = "sha256-PtscKsQwW1uBu0OfVHi7/d9JujGx12nZT+W3uMPQQWc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    cp -r $src/* $out/
```

Comment:
The result from this is:

```
bin/
build.xml
catalog-dita.xml
config/
docsrc/
integrator.xml
lib/
LICENSE
NOTICES.txt
plugins/
resources/
share/
startcmd.bat
startcmd.sh
```

This causes some (potential) problems: if one adds `dita-ot` to `environment.systemPackages`, the contents of `bin`, `lib` and `share` get symlinked to `/run/current-system/sw`. In the case of this package it means that all of the JARs in `lib/` get symlinked directly to `/run/current-system/sw/lib` and all of the directories (like `topics`, `css`, etc.) in `share/doc` get symlinked to the top level of `/run/current-system/sw/share/doc`.

I suggest you create a directory like `$out/dita-ot` or `$out/share/dita-ot`, copy the contents of the package there and then create wrappers in `$out/bin`. A (somewhat complicated) example of this is [JetBrains IDEs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/jetbrains/bin/linux.nix#L94-L95).

Another issue is that the application won’t run if you don’t have `java` in the `PATH` or `JAVA_HOME` set otherwise. You should use [`wrapProgram`](https://nixos.org/manual/nixpkgs/stable/#fun-wrapProgram) to create a wrapper that sets `JAVA_HOME` to `jdk17.home` (again, check the JetBrains example). If there are any other run-time dependencies, they should also be added to `PATH` using the wrapper.

Reply:
I guess I added it properly, but I am not perfectly sure whether it did the job. Please check whether the current state is satisfying :)